drivers/flash/spi_nor.c

```diff
@@ -500,29 +501,43 @@ static int exit_dpd(const struct device *const dev)
 
 /* Everything necessary to acquire owning access to the device.
  *
- * This means taking the lock and, if necessary, waking the device
- * from deep power-down mode.
+ * This means powering up the SPI bus, taking the lock and, if necessary,
+ * waking the device from deep power-down mode.
  */
-static void acquire_device(const struct device *dev)
+static int acquire_device(const struct device *dev)
 {
+	const struct spi_nor_config *cfg = dev->config;
+	int ret;
+
+	/* Request bus to be on. If k_sem_take blocks, the bus is powered
+	 * up anyway, so running this early has no power implications.
+	 */
+	ret = pm_device_runtime_get(cfg->spi.bus);
+	if (ret < 0) {
+		return ret;
+	}
+
 	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
 		struct spi_nor_data *const driver_data = dev->data;
 
 		k_sem_take(&driver_data->sem, K_FOREVER);
 	}
 
 	if (IS_ENABLED(CONFIG_SPI_NOR_IDLE_IN_DPD)) {
-		exit_dpd(dev);
+		ret = exit_dpd(dev);
 	}
+	return ret;
```

Comment:
Perhaps it would be good to issue some log error messages when `acquire_device()` is to return an error, so that it clearly differs from the case when the actual flash operation fails.

Reply:
Added for the case when there is an SPI bus error, failing to exit DPD is still a flash error